python/src/equistore/operations/unique_metadata.py

```diff
@@ -0,0 +1,301 @@
+    axis: str,
+    names: List[str],
+    gradient_param: Optional[str] = None,
+):
+    """
+    Checks input args for :py:func:`unique_metadata` and
+    :py:func:`unique_metadata_block`.
+    """
+    # Check tensors
+    if isinstance(tensor, TensorMap):
+        blocks = tensor.blocks()
+        # Check gradients
+        if gradient_param is not None:
+            if not isinstance(gradient_param, str):
+                raise TypeError(
+                    f"`gradient_param` must be a `str`, not {type(gradient_param)}"
+                )
+            # Check all blocks have a gradient under the passed param
+            if not np.all([block.has_gradient(gradient_param) for block in blocks]):
+                raise ValueError(
+                    "not all input blocks have a gradient under the"
+                    + f" passed `gradient_param` {gradient_param}"
+                )
+            blocks = [
+                block.gradient(gradient_param) for block in blocks
+            ]  # redefine blocks
+    elif isinstance(tensor, TensorBlock):
+        blocks = [tensor]
+        # Check gradients
+        if gradient_param is not None:
+            if not isinstance(gradient_param, str):
+                raise TypeError(
+                    f"`gradient_param` must be a `str`, not {type(gradient_param)}"
+                )
+            # Check block has a gradient under the passed param
+            if not tensor.has_gradient(gradient_param):
+                raise ValueError(
+                    "input block does not have a gradient under the"
+                    + f" passed `gradient_param` {gradient_param}"
+                )
+            blocks = [tensor.gradient(gradient_param)]  # redefine blocks
+    # Check axis
+    if not isinstance(axis, str):
+        raise TypeError(
+            "`axis` must be a `str`, either `'samples'` or `'properties'`,"
+            + f" receieved type {type(axis)}"
+        )
+    if axis not in ["samples", "properties"]:
+        raise ValueError(
+            "`axis` must be passed as either `'samples'` or `'properties'`,"
+            + f" {axis} was passed"
+        )
+    # Check names
+    if not isinstance(names, list):
+        raise TypeError(f"`names` must be a `list` of `str`, received {type(names)}")
+    if not np.all([isinstance(name, str) for name in names]):
+        raise TypeError(
+            "`names` must be a `list` of `str`, received"
+            + f" {[type(name) for name in names]}"
+        )
+    # Assumes samples/properties names are the same for every block in blocks
+    # block_names = (
+    #     blocks[0].samples.names if axis == "samples" else blocks[0].properties.names
+    # )
+    # for name in names:
+    #     if name not in block_names:
+    #         raise ValueError(
+    #             f"the block(s) passed must have {axis} names that match those"
+    #             + f" passed in `names`. {names} were passed, {block_names} found."
+    #         )
```

Comment:
why is this commented out?

Reply:
Apologies, forgot to remove. It was an old input arg check that is no longer valid because we now do not raise an error if the names are not part of the input tensor. New commit removes this!